lib/workers.js

```diff
@@ -94,10 +94,19 @@ const createWorkerObjects = (testGroups, config, testRoot, options, selectedRuns
     });
   }
   const workersToExecute = [];
+
+  const currentOutputFolder = config.output;
+  const currentMochawesomeReportDir = config.mocha.reporterOptions?.mochawesome.options.reportDir;
+
   collection.createRuns(selectedRuns, config).forEach((worker) => {
+    const _config = { ...config };
+    _config.output = `${currentOutputFolder}/${worker.config.browser.browser}`;
```

Comment:
Maybe `${worker.name}` is better than `${worker.config.browser.browser}`

Then an output folder name will be e.g. `personal:chromium1` (or `dev:chromium1`) instead of `chromium` and it's more similar to naming when using `run-multiple` mode.

The `run-multiple` mode uses `personal_chromium_1` (`<profile>_<browser_name>_<browser_order>`)

Reply:
I run a test in VSCode in debug mode and just saw the value. I guess "worker.name" returns `<profile_name>:<browser_name>1`.
Not sure for `1` means. I thought that it's a worker ID, but in my testing it's always `1` even for multiple workers (e.g. 2 workers).

Reply:
Yes, folder name is e.g. `dev_firefox1` (profile "dev", browser "firefox") after the latest fixes :heavy_check_mark: .
Also XML report `result.xml` is stored to a folder per browser :heavy_check_mark: .